tests/integration/failoverto.tcl

```diff
@@ -0,0 +1,185 @@
+            }
+
+            exec kill -SIGCONT $replica_pid
+
+            # Wait for failover to end
+            wait_for_condition 50 100 {
+                [string match *master_failover_in_progress:0* [$primary info replication]]
+            } else {
+                fail "Failover from primary to replica did not finish"
+            }
+
+            assert_match *slave* [$primary role]
+            assert_match *master* [$replica role]
+        }
+
+        test {failoverto aborts if replica never catches up} {
+            reset_replication $replica $replica_pid $primary $primary_host $primary_port
+
+            # Stop replica so it never catches up
+            exec kill -SIGSTOP $replica_pid
+            $primary SET CASE 1
+            
+            $primary failoverto any one TIMEOUT 500
+            # Wait for failover to end
+            wait_for_condition 50 20 {
+                [s 0 master_failover_in_progress] == 0
+            } else {
+                fail "Failover from primary to replica did not finish"
+            }
+
+            exec kill -SIGCONT $replica_pid
+            assert_match *master* [$primary role]
+            assert_match *slave* [$replica role]
+        }
+
+        test {failoverto aborts if primary can't sync with new replica} {
+            reset_replication $replica $replica_pid $primary $primary_host $primary_port
+
+            
+            $primary SET CASE 2
+            # We puase the primary to stop any offset accumulating before
+            # start the failover. We need to make sure the two nodes are
+            # caught and nothing is replicated in between commands. 
+            $primary client pause 1000 WRITE
+            set offset [s 0 master_repl_offset]
+
+            wait_for_condition 50 20 {
+                [string match *state=online,offset=$offset* [$primary info replication]]
+            } else {
+                fail "Primary never attempted to hand off to a replica"
+            }
+
+            # Stop the replica here, the two nodes have the same offset
+            # so the primary should demote itself and send the request here.
+            exec kill -SIGSTOP $replica_pid
+            
+            $primary failoverto any one TIMEOUT 500
+            
+            # Wait for primary to find the replica and start handoff
+            wait_for_condition 50 20 {
+                [string match *slave* [$primary role]]
+            } else {
+                fail "Primary never attempted to hand off to a replica"
+            }
+
+            # Wait for failover to end, we add 5 seconds here.
+            wait_for_condition 50 200 {
+                [s 0 master_failover_in_progress] == 0
+            } else {
+                fail "Failover from primary to replica did not finish"
+            }
+            exec kill -SIGCONT $replica_pid
+            assert_match *master* [$primary role]
+        }
+    }
+}
+
```

Comment:
I think we need to cover a case were the replica doesn't support the new FAILOVER argument for PSYNC.
I think it would be good enough to just block the PSYNC command entirely with ACL (it would be similar to getting an arity error.